sdk/js/src/EventEmitter.ts

```diff
@@ -20,10 +20,19 @@ const isInvalidEventKey = (key: string): boolean => {
     !key.startsWith(EventNames.NEW_VARIABLES)
 }
 
-export class EventEmitter {
+const callHandlers = (event: Event, handlers: eventHandler[]): void  => {
+    const args = (<CustomEvent<any[]>>event).detail
+
+    handlers.forEach((handler) => {
+        handler(...args)
+    })
+}
+
+export class EventEmitter extends EventTarget {
```

Comment:
are there any potential issues with browser compatibility that we introduce by adding this? it's a pretty new thing

Reply:
based on this https://developer.mozilla.org/en-US/docs/Web/API/EventTarget#browser_compatibility all major browsers have supported the EventTarget class since at least 2020, do you think that's good enough?

Reply:
oh wait but vercel can't build it for some reason, `(node:1454) UnhandledPromiseRejectionWarning: ReferenceError: EventTarget is not defined`
i guess because we build it with node? maybe i need to import it with yarn somehow for node

Reply:
oh no i actually think it's because vercel's version of node is 14.x and we need 14.5 or higher

Reply:
shouldn't this be running in a browser context? How come node is involved?

Reply:
yeah i'm confused about that too - i guess we build it with node? it's published to npm right?

Reply:
it is, but the build should succeed regardless. This looks more like a runtime exception from actually running it (inside Node)

Reply:
hmm i switched vercel to use node 16 for this project and it built it no problem